Libraries/LibGfx/PainterSkia.cpp

```diff
@@ -60,6 +64,198 @@ static SkPathFillType to_skia_path_fill_type(Gfx::WindingRule winding_rule)
+        auto const& solid_color = static_cast<Gfx::SolidColorPaintStyle const&>(style);
+        auto color = solid_color.sample_color(Gfx::IntPoint(0, 0));
+
+        paint.setColor(to_skia_color(color));
+    } else if (is<Gfx::CanvasLinearGradientPaintStyle>(style)) {
+        auto const& linear_gradient = static_cast<Gfx::CanvasLinearGradientPaintStyle const&>(style);
+        auto const& color_stops = linear_gradient.color_stops();
+
+        Vector<SkColor> colors;
+        colors.ensure_capacity(color_stops.size());
+        Vector<SkScalar> positions;
+        positions.ensure_capacity(color_stops.size());
+        for (auto const& color_stop : color_stops) {
+            colors.append(to_skia_color(color_stop.color));
+            positions.append(color_stop.position);
+        }
+
+        Array<SkPoint, 2> points;
+        points[0] = to_skia_point(linear_gradient.start_point());
+        points[1] = to_skia_point(linear_gradient.end_point());
+
+        SkMatrix matrix;
+        auto shader = SkGradientShader::MakeLinear(points.data(), colors.data(), positions.data(), color_stops.size(), SkTileMode::kClamp, 0, &matrix);
+        paint.setShader(shader);
+    } else if (is<Gfx::CanvasRadialGradientPaintStyle>(style)) {
+        auto const& radial_gradient = static_cast<Gfx::CanvasRadialGradientPaintStyle const&>(style);
+        auto const& color_stops = radial_gradient.color_stops();
+
+        Vector<SkColor> colors;
+        colors.ensure_capacity(color_stops.size());
+        Vector<SkScalar> positions;
+        positions.ensure_capacity(color_stops.size());
+        for (auto const& color_stop : color_stops) {
+            colors.append(to_skia_color(color_stop.color));
+            positions.append(color_stop.position);
+        }
+
+        auto start_center = radial_gradient.start_center();
+        auto end_center = radial_gradient.end_center();
+        auto start_radius = radial_gradient.start_radius();
+        auto end_radius = radial_gradient.end_radius();
+
+        auto start_sk_point = to_skia_point(start_center);
+        auto end_sk_point = to_skia_point(end_center);
+
+        SkMatrix matrix;
+        auto shader = SkGradientShader::MakeTwoPointConical(start_sk_point, start_radius, end_sk_point, end_radius, colors.data(), positions.data(), color_stops.size(), SkTileMode::kClamp, 0, &matrix);
+        paint.setShader(shader);
+    }
+}
+
+static void apply_filters(SkPaint& paint, ReadonlySpan<Gfx::Filter> filters)
+{
+    for (auto const& filter : filters) {
+        filter.visit(
+            [&](Gfx::BlurFilter blur_filter) {
+                paint.setImageFilter(SkImageFilters::Blur(blur_filter.radius, blur_filter.radius, nullptr));
+            },
+            [&](Gfx::ColorFilter color_filter) {
+                sk_sp<SkColorFilter> skia_color_filter;
+                float amount = color_filter.amount;
+
+                switch (color_filter.type) {
+                case ColorFilter::Type::Grayscale: {
```

Comment:
this code seems to be basically copy pasted from https://github.com/LadybirdBrowser/ladybird/blob/master/Libraries/LibWeb/Painting/DisplayListPlayerSkia.cpp#L136
could we come up with some helper that does conversion from Gfx::Filter to SkColorFilter and shared between PainterSkia and DisplayListPlayerSkia?

Reply:
I modified LibWeb to use `Gfx::Filters` instead of `ResolvedFilters` as they are the same. This makes most of the code shared.